.devcontainer/docker-compose.yml

```diff
@@ -0,0 +1,44 @@
+version: '3'
+
+services:
+  app:
+    build:
+      context: ..
+      dockerfile: .devcontainer/Dockerfile
+      args:
+        # Update 'VARIANT' to pick a version of Ruby: 3, 3.1, 3.0, 2, 2.7, 2.6
+        # Append -bullseye or -buster to pin to an OS version.
+        # Use -bullseye variants on local arm64/Apple Silicon.
+        VARIANT: "3.0-bullseye"
+        # Optional Node.js version to install
+        NODE_VERSION: "14"
+
+    volumes:
+      - ..:/workspace:cached
+
+    # Overrides default command so things don't shut down after the process ends.
+    command: sleep infinity
+
+    # Runs app on the same network as the database container, allows "forwardPorts" in devcontainer.json function.
+    network_mode: service:db
+    # Uncomment the next line to use a non-root user for all processes.
+    # user: vscode
+
+    # Use "forwardPorts" in **devcontainer.json** to forward an app port locally.
+    # (Adding the "ports" property to this file will not forward from a Codespace.)
+
+  db:
+    image: postgres:latest
```

Comment:
I think this'd get you a version newer than the Postgres 12 I'm using for local dev and in production and CI right now. Can we pin it to 12.x?

Reply:
well I thought this would be the easiest step, but for whatever reason the dev container fails to run if it's below postgres:14 :thonk:.

Reply:
this is now pinned to Postgres 14.

Good thing the app updated to using Postgres 14.

Why make a change when you can wait 9 months until the requirements change 😤